src/syscall/syscall_libc.go

```diff
@@ -212,6 +212,13 @@ func Truncate(path string, length int64) (err error) {
 	return
 }
 
+//go:linkname syscall_fcntl runtime/runtime.fcntl
+func syscall_fcntl(fd, cmd, arg int32) (ret int32, errno int32) {
+	// https://cs.opensource.google/go/go/+/master:src/runtime/os_linux.go;l=452?q=runtime.fcntl&ss=go%2Fgo
+	r, _, err := Syscall6(SYS_FCNTL, uintptr(fd), uintptr(cmd), uintptr(arg), 0, 0, 0)
```

Comment:
Why are you trying to do a direct system call in syscall_libc.go? `Syscall6` and the like are only supported on Windows and Linux, not on many other systems.

What about calling the libc function `fcntl` instead?

Reply:
Creating a wrapper to the libc sounds like a good idea.